tools/autograd/derivatives.yaml

```diff
@@ -1124,12 +1124,12 @@
   values: gather_with_keepdimed_indices(self_t, dim, indices, keepdim)
 
 - name: mul.Tensor(Tensor self, Tensor other) -> Tensor
-  self: mul_tensor_backward(grad, other, self.scalar_type())
-  other: mul_tensor_backward(grad, self, other.scalar_type())
+  self: mul_tensor_backward(grad, other, self, self.scalar_type())
+  other: mul_tensor_backward(grad, self, other, other.scalar_type())
```

Comment:
@soulitzer , adding these additional `self/other` arguments break `test_no_unnecessary_save` test from `test_autograd`.
A compressed repro:
```python
In [1]: import torch

In [2]: def f():
   ...:     mu = torch.ones(1, requires_grad=True)
   ...:     x = torch.empty(1)
   ...:     loss = 0
   ...:     for i in range(2):
   ...:         x.detach_()
   ...:         x.copy_(mu + i)
   ...:         ft = torch.tensor([float(i)])
   ...:         multiplied = x * ft
   ...:         s = multiplied.sum()
   ...:         loss += s
   ...:     loss.backward()
   ...: 

In [3]: f()
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [3], in <cell line: 1>()
----> 1 f()

Input In [2], in f()
     10     s = multiplied.sum()
     11     loss += s
---> 12 loss.backward()

File /work/nik/pytorch/torch/_tensor.py:482, in Tensor.backward(self, gradient, retain_graph, create_graph, inputs)
    472 if has_torch_function_unary(self):
    473     return handle_torch_function(
    474         Tensor.backward,
    475         (self,),
   (...)
    480         inputs=inputs,
    481     )
--> 482 torch.autograd.backward(
    483     self, gradient, retain_graph, create_graph, inputs=inputs
    484 )

File /work/nik/pytorch/torch/autograd/__init__.py:197, in backward(tensors, grad_tensors, retain_graph, create_graph, grad_variables, inputs)
    192     retain_graph = create_graph
    194 # The reason we repeat same the comment below is that
    195 # some Python versions print out the first line of a multi-line function
    196 # calls in the traceback and some print out the last line
--> 197 Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
    198     tensors, grad_tensors_, retain_graph, create_graph, inputs,
    199     allow_unreachable=True, accumulate_grad=True)

RuntimeError: one of the variables needed for gradient computation has been modified by an inplace operation: [torch.FloatTensor [1]], which is output 0 of torch::autograd::CopyBackwards, is at version 2; expected version 1 instead. Hint: enable anomaly detection to find the operation that failed to compute its gradient, with torch.autograd.set_detect_anomaly(True).

```
From that additional tensor I just need `sizes` and `is_sparse`. The fun part is that autograd works just fine if I pass `self.sizes()/other.sizes()` but fails with `is_sparse` with the same error message. Do you have a clue?

Reply:
We have special rules to avoid saving self when only some attributes are needed. You can add is_sparse to these patterns at https://github.com/pytorch/pytorch/blob/793deeefb44ee22dea9f6da5f57bf16e4d63d32d/tools/autograd/load_derivatives.py#L753

Reply:
Thank you, @albanD , that resolved all my issues :)